docs/resource/README.md

```diff
@@ -102,7 +102,104 @@ as specified in the [Resource SDK specification](https://github.com/open-telemet
+(in other words `service.namespace,service.name,service.instance.id` triplet MUST be globally unique).
+The ID helps to distinguish instances of the same service that exist at the same time
+(e.g. instances of a horizontally scaled service). It is preferable for the ID to be persistent
+and stay the same for the lifetime of the service instance, however it is acceptable that
+the ID is ephemeral and changes during important lifetime events for the service
+(e.g. service instance restarts).
+
+If the instance has no inherent unique ID that can be used as the value of this attribute,
+implementations MAY to generate a random Version 1 or Version 4
+[RFC 4122](https://www.ietf.org/rfc/rfc4122.txt) UUID. When enough data is available,
+implementations SHOULD use Version 5 and MUST use the following UUID as the namespace:
+`4d63009a-8d0f-11ee-aad7-4c796ed8e320`.
+
+UUIDs are typically recommended, as we only need an opaque yet reproducible value for
+the purposes of identifying a service instance. Similar to what can be seen in the man page for the
+[`/etc/machine-id`](https://www.freedesktop.org/software/systemd/man/machine-id.html) file,
+the underlying data, such as pod name and namespace should be treated as
+confidential by this algorithm, being the user's choice to expose it or not via another resource attribute.
+
+When a UUID v5 is generated, the input MUST be prefixed with
+`${telemetry.sdk.name}.${telemetry.sdk.language}.${service.namespace}.${service.name}`, followed by the
+the workload identifier, which should tentatively stable. This means that the same service yields the
+same UUID if the same identifier (`host.id`, `/etc/machine-id`, and so on) remains the same. It would still
+yield different results for different services on the same host or namespace. When no namespaces or equivalent fields
+are available, the prefix MUST then be `${telemetry.sdk.name}.${telemetry.sdk.language}.${service.name}`.
+
+Users running their services on platforms such as Kubernetes are encouraged to explicitly set the
+`service.instance.id` using their existing automation, or by setting a value that can be used for a
+consistent value, such as `container.id`. Similarly, users of application servers such as `unicorn`
+are encouraged to set the `service.instance.id` on a per-worker basis.
+
+SDKs MUST use the following algorithm when generating `service.instance.id`:
+
+- If the user has provided a `service.instance.id`, via environment
+  variable, configuration or custom resource detection, this MUST take priority over generated IDs.
+- When any of the below combinations of resource attribute are provided, they MUST be used as the input
+  for generating a UUID v5 following the prefix mentioned above. The values within each combination MUST be separated with dots:
+  * `container.id`, resulting in the input `${telemetry.sdk.name}.${telemetry.sdk.language}.${service.namespace}.${service.name}.${container.id}`
```

Comment:
Having `container.id` available at SDK runtime is very unreliable. Current resource detector implementations use /proc files that exposte that ID by accident, see https://github.com/containerd/containerd/issues/8185 for more details.

Reply:
That makes sense, to not infer it, but then I struggle a little bit with the current structure of this paragraph:

1. > Users running their services on platforms such as Kubernetes are encouraged to explicitly set the
`service.instance.id` using their existing automation [...]

2. > SDKs MUST use the following algorithm when generating `service.instance.id`: [...]
   > When any of the below combinations of resource attribute are provided, they MUST be used as the input [...]
   > * `container.id` [...]
   > * `k8s.namespace.name`/`k8s.pod.name`/`k8s.container.name` [...]

3. > When no other source is available the SDK MUST generate a value using UUID v1 or v4.
   > This would also typically be the case for service instances running on Kubernetes,

If I run K8s, I am (1) encouraged to use automation to set the id, (2) I MUST use the container.*, k8s.* values and (3) the typical case for me is that the ID is set randomly. 

I assume there is some order hidden in the intended implementation like (1) use automation and inject the ID via env, (2) use the values if available, (3) set a random value 

But it is opaque to me.

Reply:
Some thoughts:

* 1 - The _best_ is for users to specify the service.instance.id manually: they are the service owners, they know if two containers on the same pod should be treated as one service or not. 
* 1 - The Operator might in the future implement a flag to allow it to generate reliable service.instance.id, given that it has all the information necessary for that.
* 2 - Users might not want to set the whole service.instance.id on their side, but might set things like container.name via Helm. We can then use all the information we have to generate a stable service.instance.id.
* 3 - When we don't have enough information, the SDK should generate a random one

Reply:
I like this order of priorities, do you think this can be encoded in the algorithm you have outlined here?

Reply:
Done. Can you take another look?

Reply:
LGTM now, thanks for writing this down in such a comprehensive manner